presto-spark-launcher/src/main/java/com/facebook/presto/spark/launcher/PrestoSparkRunner.java

```diff
@@ -155,6 +155,7 @@ private void execute(IPrestoSparkQueryExecutionFactory queryExecutionFactory, Pr
                 prestoSparkRunnerContext.getExtraCredentials(),
                 Optional.ofNullable(prestoSparkRunnerContext.getCatalog()),
                 Optional.ofNullable(prestoSparkRunnerContext.getSchema()),
+                prestoSparkRunnerContext.getSqlText().orElse(""),
```

Comment:
I wonder if instead, we could pull out the logic for getting the sqlText from queryExecutionFactory.create() to the top of this method. And then pass in the final sql text here and change PrestoSparkQueryExecutionFactory to just take in a  non optional sqlText. That way this will get populated correctly for presto on spark.  @singcha @shrinidhijoshi would that make sense to do here?